service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -453,6 +503,9 @@ private void addLastRunDetails(List<MethodDetails> methodDetails) {
 
   private MethodDetails methodToMethodDetails(Method method, boolean includeVersions) {
 
+    Boolean isPrivate =
+        githubMethodDetailsDao.getMethodSourceDetails(method.methodId()).isPrivate();
```

Comment:
Will `.isPrivate()` throw exception if `githubMethodDetailsDao.getMethodSourceDetails(...)` returns null (for dockstore methods)?

Reply:
This would produce an exception. To handle this, I added an `if` conditional to first check if there are any method source details for the method. If this does not return null, then we set `isPrivate` to `githubMethodDetailsDao.getMethodSourceDetails(...).isPrivate()`. If the details are null, then we set `isPrivate` to null in the method details. This way, we avoid accessing `isPrivate` for dockstore methods which would throw an exception. Thank you for catching!